tools/introspection/source/introspection_app.cpp

```diff
@@ -701,43 +700,26 @@ void IntrospectionApp::runIntrospection(const iox::units::Duration updatePeriodM
         {
             prettyPrint("### MemPool Status ###\n\n", PrettyOptions::highlight);
 
-            std::deque<const MemPoolIntrospectionTopic*> mempoolSamples;
             const void* rawMempoolSample{nullptr};
 
+            while (!rawMempoolSample)
+                memPoolSubscriber.getChunk(&rawMempoolSample);
+            const MemPoolIntrospectionInfoContainer* mempoolSample =
+                static_cast<const MemPoolIntrospectionInfoContainer*>(rawMempoolSample);
 
-            while (memPoolSubscriber.getChunk(&rawMempoolSample))
-            {
-                decltype(mempoolSamples)::value_type typedMempoolSample =
-                    static_cast<decltype(mempoolSamples)::value_type>(rawMempoolSample);
-                mempoolSamples.push_back(std::move(typedMempoolSample));
-                memPoolSubscriber.releaseChunk(rawMempoolSample);
-            }
-
-            if (mempoolSamples.empty())
+            if (mempoolSample->empty())
             {
                 prettyPrint("Waiting for mempool introspection data ...\n");
             }
             else
             {
-                auto it = mempoolSamples.end() - 1;
-                auto currentIter = it;
-                auto itBegin = mempoolSamples.begin();
-
-                uint32_t lastId = (*it)->m_id;
-
-                while (it != itBegin)
+                for (const auto& i : *mempoolSample)
```

Comment:
i is actually a MempoolIntroSpectionInfo, together with renaming the container renaming i to mempoolInfo would make the code easier to read.